client/packages/common/src/utils/BarcodeScannerContext.tsx

```diff
@@ -193,9 +196,15 @@ export const BarcodeScannerProvider: FC<PropsWithChildrenOnly> = ({
     }
   };
 
+  const setScannerType = (type: ScannerType) => {
+    electronNativeAPI.setScannerType(type);
+    electronNativeAPI?.linkedBarcodeScannerDevice().then(setScanner);
```

Comment:
have to call linkedBarcodeScannerDevice, to get and set scanner (as connected or not connected, keyboardMode return connected scanner with some defaults for non optional fields of BarcodeScanner)